sql/catalyst/src/main/scala/org/apache/spark/sql/Encoders.scala

```diff
@@ -109,13 +109,29 @@ object Encoders {
    */
   def DATE: Encoder[java.sql.Date] = ExpressionEncoder()
 
+  /**
+   * Creates an encoder that serializes instances of the `java.time.LocalDate` class
+   * to the internal representation of nullable Catalyst's DateType.
+   *
+   * @since 3.0.0
+   */
+  def LOCALDATE: Encoder[java.time.LocalDate] = ExpressionEncoder()
+
   /**
    * An encoder for nullable timestamp type.
    *
    * @since 1.6.0
    */
   def TIMESTAMP: Encoder[java.sql.Timestamp] = ExpressionEncoder()
 
+  /**
+   * Creates an encoder that serializes instances of the `java.time.Instant` class
+   * to the internal representation of nullable Catalyst's TimestampType.
+   *
+   * @since 3.0.0
+   */
+  def INSTANT: Encoder[java.time.Instant] = ExpressionEncoder()
+
```

Comment:
@MaxGekk . It seems that we support these types in java bean field, too. Can we add these after line 152 of this file?
- https://github.com/apache/spark/pull/24249/files#diff-3e88c21c9270fef6eaf6f0e64ed81f27R152

Reply:
>  It seems that we support these types in java bean field, too. 

Actually I haven't checked that. I would add separate tests to `JavaBeanDeserializationSuite` together with the comment but this is not the target of the PR.

Reply:
@dongjoon-hyun I will prepare a PR for that.